src/settings/target/manifest.rs

```diff
@@ -52,27 +53,64 @@ impl Manifest {
 
     pub fn generate(
         name: String,
-        target_type: TargetType,
+        target_type: Option<TargetType>,
         config_path: &PathBuf,
         site: Option<Site>,
     ) -> Result<Manifest, failure::Error> {
+        let config_file = config_path.join("wrangler.toml");
+        let template_config_content = fs::read_to_string(&config_file);
+        let template_config = match &template_config_content {
+            Ok(content) => {
+                let config: TemplateConfig = toml::from_str(content)?;
+                config.warn_on_account_info();
+                if let Some(target_type) = &target_type {
+                    if config.target_type != *target_type {
+                        message::warn(&format!("The template recommends the \"{}\" type. Using type \"{}\" may cause errors, we recommend changing the type field in wrangler.toml to \"{}\"", config.target_type, target_type, config.target_type));
+                    }
+                }
+                Ok(config)
+            }
+            Err(err) => Err(err),
+        };
+        let template_config = match template_config {
+            Ok(config) => config,
+            Err(err) => {
+                log::info!("Error parsing template {}", err);
+                log::debug!("template content {:?}", template_config_content);
+                TemplateConfig::default()
+            }
+        };
+
+        let default_workers_dev = if let Some(route) = &template_config.route {
```

Comment:
why not just do a nested match here instead of using `if let`? I think that would be more readable

Reply:
rule of thumb I've used: if `if let` is followed by an `else`, that would be better written as a match